tests/integration/bond_test.py

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import copy
+
+import yaml
+
+from libnmstate import netapplier
+
+from .testlib import assertlib
+from .testlib import statelib
+from .testlib.statelib import INTERFACES
+
+
+BOND99_YAML_BASE = """
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    mode: balance-rr
+    slaves:
+    - eth1
+    - eth2
+"""
+
+
+def test_add_and_remove_bond_with_two_slaves():
+    state = yaml.load(BOND99_YAML_BASE)
+    netapplier.apply(copy.deepcopy(state))
```

Comment:
Is the deepcopy necessary here? If it is, wouldn't it be better to do it in netapplier.apply?

Reply:
Yes, because `apply` may change the dict we later want to compare against.
Unfortunately, we do not have a way to make sure no one changes our data.

Reply:
Ohh missed the second part. Yes, we need to move it into apply().